src/components/Searcher.js

```diff
@@ -970,6 +990,44 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
     });
   }
 
+  /**
+   * Add signal each time a filter is added/removed.
+   * When a filter is added, signal with weight 1 is created.
+   * When a filter is removed, signal with weight 0 is created.
+   */
+  addSignal(facetFilter: FacetFilter, weight: number) {
+    const { query, queryTimestamp } = this.state;
+    const savedUser = AuthUtils.getSavedUser();
+    const facets = this.state.response ? this.state.response.facets : null;
+    if (!savedUser || !facets) {
+      return;
+    }
+    const facet = facets.find((searchFacet) => {
+      return searchFacet.label === facetFilter.facetName;
+    });
+    if (!facet) {
+      return;
+    }
+    const signal = new SignalData();
+    signal.docId = facetFilter.filter;
+    signal.docOrdinal = facet.buckets.findIndex((bucket) => {
```

Comment:
If the query returned documents, I'm wondering if we should try to get most of this information out of the signal returned for the first document. This will provide much more accurate information and allow tools to better associate these facet filter signals with others. The query, queryTimestamp, relevancyModelName, etc. fields, for example, will be different than what you can get from the searcher...

What do you think about this idea?

Reply:
Would be nice to use that. But which fields would be relevant from that signal? Only ones I can think of are locale, principal.

Reply:
Actually, I think a lot of them would be...
* principal
* query
* queryTimestamp
* relevancyModelName
* relevancyModelNames

And maybe locale, too...

For example, the query field could be different than the string that the searcher knows about because the back-end code can transform it and using the one that will end up in click signals will make the click signals and facet signals easier to match up later. The same for the queryTimestamp.... knowing the queryTimestamp from the original query will also help match things up.

We could save these fields instead of just the query time you're currently saving (since that won't match the one in the signals anyway). If there are no documents returned, then there won't be any applicable facet filters, so we should always have the data available when submitting a facet signal, I believe.

Reply:
Oh yeah, that makes sense. For some reason I thought the query (and therefore queryTimestamp) would change when we apply filter. But my bad.
Looks like most of the properties from querySignal can be easily used here.